src/column.rs

```diff
@@ -582,7 +581,7 @@ impl Column {
 					(size_tier, offset)
 				};
 
-				if skip_preimage_indexes && self.preimage && size_tier != 15 {
+				if skip_preimage_indexes && self.preimage && size_tier as usize != tables.value.len() - 1 {
```

Comment:
Actually here also I was thinking of '!= tables.value.len()'

Reply:
`tables` here also includes the multipart table